File: 0189-rotate-array/solution.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
int n=nums.size();

          if (n <= 1) return;       // nothing to rotate
    k = k % n;                // handle k > n
    if (k == 0) return; 

        vector<int> temp(n);
        int a=0;
        for(int i=nums.size()-k;i<nums.size();i++){

            temp[a]=nums[i];
            a++;

        }

              for(int i=0;i<nums.size()-k;i++){

            temp[a]=nums[i];
            a++;

        }

        for(int i=0;i<nums.size();i++){
            nums[i]=temp[i];
        }
        
    }
};

File: 0189-rotate-array/solution.java
class Solution {

    void rev(int arr[],int s,int e){
        int i=s;
        int j=e;

        while(i<e){
            int temp=arr[i];
            arr[i] = arr[e];
            arr[e] = temp;

            i++;
            e--;
        }
    }

    public void rotate(int[] nums, int k) {

        int size = nums.length;

        if(k>size) k=k%size;


        rev(nums,0,size-1);
        rev(nums,0,k-1);
        rev(nums,k,size-1);

        
    }
}
